spec/compiler/interpreter/class_vars_spec.cr

```diff
@@ -126,5 +126,23 @@ describe Crystal::Repl::Interpreter do
         a
       CODE
     end
+
+    it "finds self in class var initializer (#12439)" do
+      interpret(<<-CODE).should eq(42)
```

Comment:
My comment is partly off-topic because it's very general and can't immediately be solved in this PR.

I've been thinking about how interpreter specs are being handled, and something keeps bothering me:

Adding such a spec to *just* the interpreter seems wrong to me. This just describes behavior of Crystal language; just because the interpreter happened to have it wrong, doesn't mean it should be an interpreter-only spec. It would also be just as valuable for normal compiler. *And* it would be trivial to just reuse this exact code snippet and expectation and apply it to the general compiler.

Going further, it seems that all specs that look like this (code snippet + expected result) can and should be defined in one place and run for both the compiler and interpreter. When doing that, we will likely see that many of such specs actually fail on the interpreter in the current state. But, isn't that a good thing? What if the compiler actually already has a spec that would show the same failure and so we wouldn't need to wait until some user happens to run into this issue?

Reply:
Good point!

Maybe an initial way to do this would be having a helper method that runs some code in compiled and interpreted mode? Then we can move more and more specs to be like that, until eventually all or most of them run in that way, and then that could be the default way to run things.